plugins/services/src/js/validators/MarathonAppValidators.js

```diff
@@ -324,6 +324,34 @@ const MarathonAppValidators = {
 
       return errors;
     }, []);
+  },
+
+  /**
+   * @param {Object} app - The data to validate
+   * @returns {Array} Returns an array with validation errors
+   */
+  validateLabels(app) {
+    if (ValidatorUtil.isDefined(app.labels)) {
+      return Object.keys(app.labels)
+        .reduce(function(accumulator, labelKey) {
+          if (/^\s|\s$/.test(labelKey)) {
+            accumulator.push(`labels.${labelKey}`);
+          }
+
+          return accumulator;
+        }, [])
+        .map(function(labelPath) {
+          return {
+            path: [labelPath],
+            message: "Keys must not start or end with whitespace characters",
```

Comment:
I decided not to because the rest of the file isn't using it. But if you think we should do it for this case, I'll be happy to do it!